File: CS/WebSite/Models/Person.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace DXWebApplication3.Models {
    public class Person {
        [Key]
        public int ID { get; set; }
        public bool CanSpeakReceived { get; set; }
        public bool CanWriteReceived { get; set; }
        public bool CanUnderstandReceived { get; set; }
        public bool CanSpeakExpressed { get; set; }
        public bool CanWriteExpressed { get; set; }
        public bool CanUnderstandExpressed { get; set; }
        public Person() { }
    }
    public static class PersonLanguages {
        public static List<Person> GetPersonLanguages() {
            if (HttpContext.Current.Session["ModelList"] == null) {
                List<Person> result = new List<Person>();
                result.Add(new Person() {
                    ID = 0,
                    CanSpeakExpressed = true,
                    CanSpeakReceived = true,
                    CanUnderstandExpressed = true,
                    CanUnderstandReceived = false,
                    CanWriteExpressed = false,
                    CanWriteReceived = false
                });
                result.Add(new Person() {
                    ID = 1,
                    CanSpeakExpressed = false,
                    CanSpeakReceived = true,
                    CanUnderstandExpressed = false,
                    CanUnderstandReceived = true,
                    CanWriteExpressed = false,
                    CanWriteReceived = true
                });
                result.Add(new Person() {
                    ID = 2,
                    CanSpeakExpressed = true,
                    CanSpeakReceived = true,
                    CanUnderstandExpressed = false,
                    CanUnderstandReceived = true,
                    CanWriteExpressed = false,
                    CanWriteReceived = false
                });
                HttpContext.Current.Session["ModelList"] = result;
            }
            return (List<Person>)HttpContext.Current.Session["ModelList"];
        }

        public static List<Person> UpdatePerson(Person modelInfo) {
            Person editedModel = GetPersonLanguages().Where(m => m.ID == modelInfo.ID).First();
            editedModel.CanSpeakExpressed = modelInfo.CanSpeakExpressed;
            editedModel.CanSpeakReceived = modelInfo.CanSpeakReceived;
            editedModel.CanUnderstandExpressed = modelInfo.CanUnderstandExpressed;
            editedModel.CanWriteExpressed = modelInfo.CanWriteExpressed;
            editedModel.CanWriteReceived = modelInfo.CanWriteReceived;
            editedModel.CanUnderstandReceived = modelInfo.CanUnderstandReceived;
            return GetPersonLanguages();
        }
    }
}

File: VB/WebSite/Models/Person.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.Web

Namespace DXWebApplication3.Models
	Public Class Person
		Private privateID As Integer
		<Key> _
		Public Property ID() As Integer
			Get
				Return privateID
			End Get
			Set(ByVal value As Integer)
				privateID = value
			End Set
		End Property
		Private privateCanSpeakReceived As Boolean
		Public Property CanSpeakReceived() As Boolean
			Get
				Return privateCanSpeakReceived
			End Get
			Set(ByVal value As Boolean)
				privateCanSpeakReceived = value
			End Set
		End Property
		Private privateCanWriteReceived As Boolean
		Public Property CanWriteReceived() As Boolean
			Get
				Return privateCanWriteReceived
			End Get
			Set(ByVal value As Boolean)
				privateCanWriteReceived = value
			End Set
		End Property
		Private privateCanUnderstandReceived As Boolean
		Public Property CanUnderstandReceived() As Boolean
			Get
				Return privateCanUnderstandReceived
			End Get
			Set(ByVal value As Boolean)
				privateCanUnderstandReceived = value
			End Set
		End Property
		Private privateCanSpeakExpressed As Boolean
		Public Property CanSpeakExpressed() As Boolean
			Get
				Return privateCanSpeakExpressed
			End Get
			Set(ByVal value As Boolean)
				privateCanSpeakExpressed = value
			End Set
		End Property
		Private privateCanWriteExpressed As Boolean
		Public Property CanWriteExpressed() As Boolean
			Get
				Return privateCanWriteExpressed
			End Get
			Set(ByVal value As Boolean)
				privateCanWriteExpressed = value
			End Set
		End Property
		Private privateCanUnderstandExpressed As Boolean
		Public Property CanUnderstandExpressed() As Boolean
			Get
				Return privateCanUnderstandExpressed
			End Get
			Set(ByVal value As Boolean)
				privateCanUnderstandExpressed = value
			End Set
		End Property
		Public Sub New()
		End Sub
	End Class
	Public NotInheritable Class PersonLanguages
		Private Sub New()
		End Sub
		Public Shared Function GetPersonLanguages() As List(Of Person)
			If HttpContext.Current.Session("ModelList") Is Nothing Then
				Dim result As New List(Of Person)()
				result.Add(New Person() With {.ID = 0, .CanSpeakExpressed = True, .CanSpeakReceived = True, .CanUnderstandExpressed = True, .CanUnderstandReceived = False, .CanWriteExpressed = False, .CanWriteReceived = False})
				result.Add(New Person() With {.ID = 1, .CanSpeakExpressed = False, .CanSpeakReceived = True, .CanUnderstandExpressed = False, .CanUnderstandReceived = True, .CanWriteExpressed = False, .CanWriteReceived = True})
				result.Add(New Person() With {.ID = 2, .CanSpeakExpressed = True, .CanSpeakReceived = True, .CanUnderstandExpressed = False, .CanUnderstandReceived = True, .CanWriteExpressed = False, .CanWriteReceived = False})
				HttpContext.Current.Session("ModelList") = result
			End If
			Return CType(HttpContext.Current.Session("ModelList"), List(Of Person))
		End Function

		Public Shared Function UpdatePerson(ByVal modelInfo As Person) As List(Of Person)
			Dim editedModel As Person = GetPersonLanguages().Where(Function(m) m.ID = modelInfo.ID).First()
			editedModel.CanSpeakExpressed = modelInfo.CanSpeakExpressed
			editedModel.CanSpeakReceived = modelInfo.CanSpeakReceived
			editedModel.CanUnderstandExpressed = modelInfo.CanUnderstandExpressed
			editedModel.CanWriteExpressed = modelInfo.CanWriteExpressed
			editedModel.CanWriteReceived = modelInfo.CanWriteReceived
			editedModel.CanUnderstandReceived = modelInfo.CanUnderstandReceived
			Return GetPersonLanguages()
		End Function
	End Class
End Namespace